### PR TITLE
Sort by Recent or Popular on the user page should only appear for artists and not non-artist users

### PR DESCRIPTION
**Pull request by sddioulde**
### Description
Sort by Recent / Popular on the user page should only appear for artists and not non-artist users. This PR checks whether the user is an artist, which has been updated to mean whether the user has uploaded tracks.

### Dragons

Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?
N/A

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.
Tested locally

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -78,7 +78,7 @@ npm run test
 
 ```bash
 npm run lint  # Show lint errors
-npm run lint-fix  # Fix lint errors
+npm run lint:fix  # Fix lint errors
```
Comment (raymondjacobson) [inline README.md:81]: you are the MAN!

**Comment by audius-infra** [comment]
Preview this change https://frontend-demo.audius.co/saliou-sort-recent-popular-234

File: package.json [hunk 2]
```diff
@@ -102,7 +102,7 @@
     "start:mobile-stage": "env-cmd ./.env/.env.mobile npm run start:stage",
     "prebuild": "npm run publish-scripts",
     "build": "react-scripts --max_old_space_size=8192 build",
-    "build:dev": "npm run write-sha && env-cmd ./.env/.env.dev env-cmd ./.env/.env.git npm run build && rm -rf build-staging && mv build build-development",
+    "build:dev": "npm run write-sha && env-cmd ./.env/.env.dev env-cmd ./.env/.env.git npm run build && rm -rf build-development && mv build build-development",
```
Comment (piazzatron) [inline package.json:105]: 😮

File: src/components/general/NavBanner.js [hunk 3]
```diff
@@ -29,15 +29,14 @@ const NavBanner = props => {
...
-                variant='border'
-                menu={menu}
-                disabled={props.dropdownDisabled}
-              />
-            )}
-          </div>
+
+          {props.isArtist && (
+            <div className={styles.dropdown}>
+              {!props.dropdownDisabled && (
```
Comment (piazzatron) [inline src/components/general/NavBanner.js:35]: Nice simplification! Do we want to still render the outer dropdown container if dropdownDisabled?

File: src/containers/profile-page/ProfilePageProvider.tsx [hunk 4]
```diff
@@ -620,7 +620,7 @@ class ProfilePage extends PureComponent<ProfilePageProps, ProfilePageState> {
 
   getIsArtist = () => {
     const { profile } = this.props.profile
-    return !!profile && profile.is_creator && profile.track_count > 0
+    return !!profile && profile.track_count > 0
```
Comment (piazzatron) [inline src/containers/profile-page/ProfilePageProvider.tsx:623]: FYI, we haven't really settled on a convention for this but we can also write profile?.track_count > 0 ?? false if you prefer

**Review by piazzatron — APPROVED** [review]
Epic 🛳

File: src/components/general/NavBanner.js (hunk 3 above)
Comment (sddioulde) [reply src/components/general/NavBanner.js:35]: This is a great question that also had me wondering. We can circle back to it and merge the conditions if needed. Would be cleaner for sure as long as it makes sense functionally :)

File: src/containers/profile-page/ProfilePageProvider.tsx (hunk 4 above)
Comment (sddioulde) [reply src/containers/profile-page/ProfilePageProvider.tsx:623]: Nice!
I think we have a nice opportunity here to discuss some coding style. I'm usually open to most things. We can decide whether we want to converge to one style gradually, or diverge. Less headache with the former, more fun with the latter. Let's discuss!

**Review by raymondjacobson — APPROVED** [review]
pro-tip, i think you need to have aud-234 instead of just 234 written down for linear to automatically comment :)

**Pull request merged**